pkg/workloadmeta/collectors/internal/remote/process_collector/process_collector_test.go

```diff
@@ -273,19 +268,21 @@ func TestCollection(t *testing.T) {
 			mockStore.Notify(test.preEvents)
 
 			ctx, cancel := context.WithCancel(context.TODO())
-			// Start collection
-			err = collector.Start(ctx, mockStore)
-			require.NoError(t, err)
 
+			// Subscribe to the mockStore
 			ch := mockStore.Subscribe(dummySubscriber, workloadmeta.NormalPriority, nil)
 			doneCh := make(chan struct{})
 
-			numberOfReponse := len(test.serverResponses)
-			if test.errorResponse {
-				numberOfReponse++
-			}
+			<-ch // Wait that the initial empty events are sent
+
+			// Collect process data
+			err = collector.Start(ctx, mockStore)
+			require.NoError(t, err)
+
+			numberOfEvents := len(test.serverResponses)
+
 			go func() {
```

Comment:
Why do we fork a child goroutine here ?
It seems that the parent goroutine does nothing but waiting for this goroutine to finish.
The only thing that this goroutine can execute in parallel with the parent one is `mockStore.Unsubscribe(ch)`.
Is it important to have `mockStore.Unsubscribe(ch)` running in a background goroutine when `mockStore.GetProcess(…)` is invoked in the parent goroutine line 301 ?

Reply:
I think we don't need one. IIRC I forgot to remove it after refactoring the test but it helped to catch a data race in `Unsubscribe()`